### PR TITLE
Optimize observed holidays, part 2

### PR DESCRIPTION
**Pull request by KJhellico**
Angola, Eswatini, Kazakhstan, Kenya, Malawi, Singapore, South Africa

### CODE REVIEW

File: holidays/countries/kenya.py [hunk 1]
```diff
@@ -26,29 +26,58 @@ class Kenya(HolidayBase):
     """
 
     country = "KE"
+    special_holidays = {
+        2020: ((FEB, 11, "President Moi Celebration of Life Day	"),),
```
Comment (PPsyrius) [inline holidays/countries/kenya.py:30]: Remove extra space here:
```suggestion
        2020: ((FEB, 11, "President Moi Celebration of Life Day"),),
```

File: holidays/countries/south_africa.py [hunk 2]
```diff
@@ -77,35 +82,27 @@ def _populate(self, year):
...
+            _add_with_observed(date(year, MAR, 21), "Human Rights Day")
+            _add_with_observed(date(year, APR, 27), "Freedom Day")
+            _add_with_observed(date(year, MAY, 1), "Workers' Day")
+            _add_with_observed(date(year, JUN, 16), "Youth Day")
+            _add_with_observed(date(year, AUG, 9), "National Women's Day")
+            _add_with_observed(date(year, SEP, 24), "Heritage Day")
+
+        # Special holiday http://tiny.cc/za_y2k
+        if self.observed and year == 2000:
+            self[date(2000, JAN, 3)] = "Y2K changeover (Observed)"
```
[the remark this answers is not in the document]
Comment (PPsyrius) [reply holidays/countries/south_africa.py:105]: Here's the one for the special holidays section:
```python
        2000: (
            (JAN, 2, "Y2K changeover"),
            (JAN, 3, "Y2K changeover (Observed)"),
        ),
```

File: holidays/countries/singapore.py [hunk 3]
```diff
@@ -268,30 +277,20 @@ def _add_holiday(dt: date, hol: str) -> None:
...
-            for hol_date, hol_name in list(self.items()):
-                if hol_date.year == year and self._is_sunday(hol_date):
-                    in_lieu_date = hol_date + td(days=+1)
-                    while in_lieu_date.year == year and in_lieu_date in self:
-                        in_lieu_date += td(days=+1)
-                    _add_holiday(in_lieu_date, f"{hol_name} (Observed)")
-
         # special case (observed from previuos year)
         if self.observed and year == 2007:
             self[date(2007, JAN, 2)] = "Hari Raya Haji (Observed)"
```
Comment (PPsyrius) [inline holidays/countries/singapore.py:296]: This one could also be declared in the special holidays section instead
```suggestion
        # special case (observed from previuos year)
        if self.observed and year == 2007:
            self[date(2007, JAN, 2)] = "Hari Raya Haji (Observed)"
```

Comment (PPsyrius) [reply holidays/countries/singapore.py:296]:
```python
        2007: ((JAN, 2, "Hari Raya Haji (Observed)"),),
```

**Review by PPsyrius — CHANGES_REQUESTED** [review]
Seems good IMO, this one can be improved though, now that we're touching them anyway.

**Review by arkid15r — COMMENTED** [review]
This is a huge great-quality work! Thank you for optimizing the performance issues!

LGTM

**Comment by arkid15r** [comment]
Great stuff 👍👍

**Pull request merged**